packages/constants/src/constants.ts

```diff
@@ -1,38 +1,122 @@
+// DOM elements
 export const PORTALS_CONTAINER_ID = 'portals-container';
 
+// Links
 export const SUPPORT_PORTAL_URL =
   'https://jira.commercetools.com/servicedesk/customer/portal/1/create/167';
 
-export const __GLOBAL = 'GLOBAL';
-
-// NOTIFICATIONS
+// Notification actions
 export const SHOW_LOADING = 'SHOW_LOADING';
 export const HIDE_LOADING = 'HIDE_LOADING';
 export const HIDE_ALL_PAGE_NOTIFICATIONS = 'HIDE_ALL_PAGE_NOTIFICATIONS';
-export const DOMAINS = {
+
+// Notifications
+export type TAppNotificationDomain = 'global' | 'page' | 'side';
+export const DOMAINS: {
+  [key: string]: TAppNotificationDomain;
+} = {
   GLOBAL: 'global',
   PAGE: 'page',
   SIDE: 'side',
 };
 
-// NO VALUE FALLBACK
+export type TAppNotificationKind =
+  | 'error'
+  | 'warning'
+  | 'info'
+  | 'success'
+  | 'api-error'
+  | 'unexpected-error';
+export type TAppNotificationKindSide = Omit<
+  TAppNotificationKind,
+  'api-error' | 'unexpected-error'
+>;
+export type TAppNotificationKindGlobal = Omit<
+  TAppNotificationKind,
+  'unexpected-error'
+>;
+export type TAppNotificationKindPage = TAppNotificationKind;
+export type TAppNotificationOfDomain = { domain: TAppNotificationDomain };
+export type TAppNotificationOfKind<
+  T extends TAppNotificationOfDomain
+> = TAppNotificationOfDomain & {
+  // Conditionally check the shape based on other values
+  kind: 'global' extends T['domain']
```

Comment:
The idea here is that we specify the type of `kind` based on the `domain` value.